doc/indexing.rst

```diff
@@ -353,67 +331,129 @@ labels of the 1st and 2nd dimension, respectively. You must specify all
+    da
+    da[[0, 1], [1, 1]]
+    # Sequential indexing gives the same result.
+    da[[0, 1], [1, 1]] == da[[0, 1]][:, [1, 1]]
 
-.. _SettingWithCopy warnings: http://pandas.pydata.org/pandas-docs/stable/indexing.html#returning-a-view-versus-a-copy
 
-.. _orthogonal:
+In order to make more advanced indexing, you can supply
+:py:meth:`~xarray.DataArray` as indexers.
+In this case, the dimension of the resultant array is determined
+by the indexers' dimension names,
+
+.. ipython:: python
+
+    ind_x = xr.DataArray([0, 1], dims=['x'])
+    ind_y = xr.DataArray([0, 1], dims=['y'])
+    da[ind_x, ind_y]  # orthogonal indexing
+    da[ind_x, ind_x]  # vectorized indexing
+
+If you just provide slices or sequences, which do not have named-dimensions,
+they will be understood as the same dimension which is indexed along.
+
+.. ipython:: python
+
+    # Because [0, 1] is used to index along dimension 'x',
+    # it is assumed to have dimension 'x'
+    da[[0, 1], ind_x]
+
+
+Furthermore, you can use multi-dimensional :py:meth:`~xarray.DataArray`
+as indexers, where the resultant array dimension is also determined by
+indexers' dimension,
+
+.. ipython:: python
+
+    ind = xr.DataArray([[0, 1], [0, 1]], dims=['a', 'b'])
+    da[ind]
+
+To summarize, our advanced indexing is based on our broadcasting scheme.
+See :ref:`compute.broadcasting` for the detail.
 
-Orthogonal (outer) vs. vectorized indexing
-------------------------------------------
 
-Indexing with xarray objects has one important difference from indexing numpy
-arrays: you can only use one-dimensional arrays to index xarray objects, and
-each indexer is applied "orthogonally" along independent axes, instead of
-using numpy's broadcasting rules to vectorize indexers. This means you can do
-indexing like this, which would require slightly more awkward syntax with
-numpy arrays:
+These advanced indexing also works with ``isel``, ``loc``, and ``sel``.
 
 .. ipython:: python
 
-    arr[arr['time.day'] > 1, arr['space'] != 'IL']
+    ind = xr.DataArray([[0, 1], [0, 1]], dims=['a', 'b'])
+    da.isel(y=ind)  # same to da[:, ind]
+
+    ind = xr.DataArray([['a', 'b'], ['b', 'a']], dims=['a', 'b'])
+    da.loc[:, ind]  # same to da.sel(y=ind)
+
+
+and also for Dataset
+
+.. ipython:: python
+
+    ds2 = da.to_dataset(name='bar')
+    ds2.isel(x=xr.DataArray([0, 1, 2], dims=['points']))
+
+.. note::
+  This advanced indexing was newly added in v.0.10.
+  In the older version of xarray, dimensions of indexers are not used.
+  Special methods to realize some advanced indexing,
+  ``isel_points`` and ``sel_points`` are now deprecated.
 
-This is a much simpler model than numpy's `advanced indexing`__. If you would
-like to do advanced-style array indexing in xarray, you have several options:
 
-* :ref:`pointwise indexing`
-* :ref:`masking with where`
-* Index the underlying NumPy array directly using ``.values``, e.g.,
+More advanced indexing
```

Comment:
Let's link to this section specifically when from "What's New" when we mention `isel_points` and `sel_points` are deprecated.